src/rectangle.ml

```diff
@@ -86,3 +86,38 @@ let empty =
   Array.Floatarray.unsafe_set arr 2 0.;
   Array.Floatarray.unsafe_set arr 3 0.;
   arr
+
+let get arr i = List.nth arr i
+
+let mindist p e =
+  let x0, y0, x1, y1 = (x0 e, y0 e, x1 e, y1 e) in
+  let gete ind = match ind with 0 -> x0 | 1 -> y0 | 2 -> x1 | _ -> y1 in
```

Comment:
I think these indirections are potentially a little more confusing than helpful. I think it would be clearer and more maintainable to write these in full, even if it seems a little more verbose. Seem my overview comment before making changes to this directly though.